src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1877,6 +1910,19 @@ subroutine gsi_fv3ncdf2d_read(fv3filenamegin,it,ges_z,ges_t2m,ges_q2m)
           iret=nf90_inquire_dimension(gfile_loc,k,name,len)
           dim(k)=len
        enddo
+
+!---   check the existence of significant wave height (howv) in 2D FV3-LAM firstguess file
+       if ( i_howv_in_anav == 1 ) then
+         iret_howv = nf90_inq_varid(gfile_loc,'howv',id_howv)
+         if ( iret_howv /= nf90_noerr ) then
+           i_howv_in_data = 0                ! howv does not exist in firstguess
+           write(6,'(1x,A,1x,A,1x,A,1x,I4,1x,A,1x,I4.4,A)') 'subroutine gsi_fv3ncdf2d_read:: howv is NOT found in firstguess ', trim(sfcdata), ', iret = ',iret_howv, ' (on pe: ', mype,')'
+         else
+           i_howv_in_data = 1                ! howv is found in firstguess
+           write(6,'(1x,A,1x,A,1x,A,1x,I4,1x,I4,1x,A,1x,I4.4,A)') 'subroutine gsi_fv3ncdf2d_read:: Found howv in firstguess ', trim(sfcdata), ', iret, varid = ',iret_howv, id_howv,' (on pe: ', mype,')'
```

Comment:
These lines are done only on the PE which reads the surface firstguess, not by all the PEs. So only one PE prints this message. In the real run if GSI, it ddid show just one line in the "stdout" log file.

Reply:
I understand that for regional FV3 netcdf files, we use parallel I/O. You may check with @TingLei-NOAA whether parallel I/O applies to howv (if not yet, we may want to get it done in parallel, which will benefit the NA_3km domain)

Reply:
@guoqing-noaa Thanks for bringing out this point.  For being now, in the parallel io for fv3-lam,  these 2d fields are not considered ,which are still treated by subs like gsi_fv3net2d_read.  In the future,  when those 2d fields are more and more in the future, we may need to consider a similar parallelization to them.

Reply:
@TingLei-NOAA @guoqing-noaa 
Thank you, Ting, for the explanation on this point.

Reply:
@TingLei-NOAA Thanks for the explanation. I think howv maybe the first 2D field to be read by regional GSI? 
@GangZhao-NOAA it will be good if we can get the information about what extra minutes are needed to read the extra howv field for NA_3km domain. Thanks!

Reply:
@guoqing-noaa  there are a group of 2d fields to be read in in gsi_fv3ncf2d_read and somes are written out through gsi_fv3ncdf_write_sfc in the current gsi.

Reply:
@TingLei-NOAA There is a subroutine gsi_fv3ncdf2d_read_v1(...) but it is never called anywhere. Do you have any idea what it is for? Thanks!

Reply:
@guoqing-noaa That is originally supposed to read some 2d fields from cold start files, while never actually implemented and used. It could be removed safely.

Reply:
@guoqing-noaa @TingLei-NOAA 
As Ting answered, there are a bunch of 2d fields read from 2-D firstguess file (sfc_data.nc), including t2m, q2m and howv.

As for the "information about what extra minutes are needed to read the extra howv field for NA_3km domain", to be honest, I do not know. But the adding the analysis of howv (including reading firstguess, output analysis, read obs and cost function, minimisation, etc) does not increase the time cost much, even for 3km NA domain. 
With my testing case on 3km NA domain, on Hera hybrid run (with gdas ensemble) without howv the wall-clock time is 973 seconds,  with howv it is 974 seconds.